### PR TITLE
Remove slj

### LINKED ISSUE #1181: Remove SLJ
## Description

Remove the SLJ pair potential.

## Motivation and context

The new ExpandedLJ potential in #1152 replaces SLJ. Remove SLJ in the next beta release after adding ExpandedLJ.

### PR DESCRIPTION
**Pull request by cbkerr**
## Description

What it says on the tin.

TODO remains in dem/pair.py. I'll highlight with a comment.

## Motivation and context

Resolves #1181 

## How has this been tested?
I removed tests for it!

## Change log

```
[breaking] Removed md.pair.SLJ potential and wall. Use md.pair.ExpandedLJ instead.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/dem/pair.py [hunk 1]
```diff
@@ -295,6 +295,9 @@ class SWCA(hoomd.md.force._force, _DEMBase):
     :math:`r_{cut}=2^{1/6}\sigma` and :math:`\sigma=2\cdot r` were
     applied between the contact points of each pair of particles.
 
+    TODO::
```
Comment (cbkerr) [inline hoomd/dem/pair.py:298]: Note how above this it states:
"The effect is as if a
    :py:class:`hoomd.md.pair.SLJ` interaction with
    :math:`r_{cut}=2^{1/6}\sigma` and :math:`\sigma=2\cdot r` were
    applied between the contact points of each pair of particles."

File: hoomd/md/EvaluatorPairExpandedLJ.h [hunk 2]
```diff
@@ -38,8 +38,8 @@ namespace md
 
     EvaluatorPairExpandedLJ evaluates the function:
     \f{eqnarray*}
-    V_{\mathrm{SLJ}}(r)  = & 4 \varepsilon \left[ \left( \frac{\sigma}{r - \Delta} \right)^{12} -
```
Comment (tommy-waltmann) [inline hoomd/md/EvaluatorPairExpandedLJ.h:41]: Is this something we missed when reviewing the `ExpandedLJ` PR?

File: hoomd/md/nlist.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ class NList(_HOOMDBaseObject):
 
     .. rubric:: Diameter shifting
 
-    Set `diameter_shift` to `True` when using `hoomd.md.pair.SLJ` or
+    Set `diameter_shift` to `True` when using
```
Comment (tommy-waltmann) [inline hoomd/md/nlist.py:65]: Re-organize the lines in this paragraph so the end of each line is close to the 80 character limit.

**Review by tommy-waltmann — CHANGES_REQUESTED** [review]
Good work on this, its close to being ready to merge. I have two additional comments.

I did a grep through the codebase, and there are still references to `EvaluatorPairSLJ` in the file `md/PotentialExternalGPU.cu`.

Since we are removing `SLJ` walls, are there plans to have a new type of wall that can replicate the old behavior in v3 such as having `ExpandedLJ` walls?

File: hoomd/dem/pair.py (hunk 1 above)
Comment (joaander) [reply hoomd/dem/pair.py:298]: DEM is being replaced by ALJ in v3.0.

**Comment by joaander** [comment]
> Since we are removing `SLJ` walls, are there plans to have a new type of wall that can replicate the old behavior in v3 such as having `ExpandedLJ` walls?

Yes, users can place their walls appropriately. A shift in `r` by `Delta` is identical to shifting the wall by `-Delta`.

File: hoomd/dem/pair.py (hunk 1 above)
Comment (cbkerr) [reply hoomd/dem/pair.py:298]: Should I delete this comparison?

File: hoomd/md/EvaluatorPairExpandedLJ.h (hunk 2 above)
Comment (cbkerr) [reply hoomd/md/EvaluatorPairExpandedLJ.h:41]: Yes

File: hoomd/md/nlist.py (hunk 3 above)
Comment (cbkerr) [reply hoomd/md/nlist.py:65]: Okay! I thought one of the automatic hooks would do that

**Comment by cbkerr** [comment]
> md/PotentialExternalGPU.cu

Good catch, I only grepped the c files :/

File: hoomd/md/nlist.py (hunk 3 above)
Comment (joaander) [reply hoomd/md/nlist.py:65]: flake8 will tell you when you go over the limit. It does not report lines that are under the limit or reflow paragraphs.

**Review by tommy-waltmann — APPROVED** [review]
Thanks for making those changes.

File: hoomd/md/nlist.py [hunk 4]
```diff
@@ -62,12 +62,12 @@ class NList(_HOOMDBaseObject):
...
-    interact under the modified :math:`r_\mathrm{cut}` conditions in those
-    potentials. When `diameter_shift` is `True`, set `max_diameter` to the
-    largest value that any particle's diameter will achieve (where **diameter**
-    is the per particle quantity stored in the `hoomd.State`).
+    Set `diameter_shift` to `True` when using `hoomd.md.pair.DLVO` so that the
+    neighbor list includes all particles that interact under the modified
+    :math:`r_\mathrm{cut}` conditions in those potentials. When `diameter_shift`
+    is `True`, set `max_diameter` to the largest value that any particle's
+    diameter will achieve (where **diameter** is the per particle quantity
+    stored in the `hoomd.State`).
```
Comment (joaander) [inline hoomd/md/nlist.py:70]: Remove the whole "Diameter shifting" section. No pair potentials remain that use it. We don't need to do anything about the implementation now, that can be in another PR.

File: hoomd/state.py [hunk 5]
```diff
@@ -122,7 +122,7 @@ class State:
     - :math:`d`: diameter :math:`[\\mathrm{length}]` - deprecated in v3.0.0.
       HOOMD-blue reads and writes particle diameters, but does not use them in
       any computations. As of the current beta release, diameter is still used
-      in `md.pair.SLJ` and `md.pair.DLVO`.
+      in `md.pair.DLVO`.
```
Comment (joaander) [inline hoomd/state.py:125]: Remove the "As of the current beta release" sentence. DLVO no longer uses diameters.

**Review by joaander — CHANGES_REQUESTED** [review]
I apparently failed to remove references to DLVO using diameter when I switched DLVO to use Delta. Some blocks of documentation can be removed (see inline comments).

**Review by joaander — APPROVED** [review]
Thanks!
